web/client-api/src/test/java/io/deephaven/web/client/api/HierarchicalTableTestGwt.java

```diff
@@ -505,4 +507,157 @@ public void testCreateRollupsConstituents() {
+
+                                    assertEquals(41L, ((LongWrapper) yPlusAgg.get(row0)).getWrapped());
+                                    assertEquals(14L, ((LongWrapper) yPlusAgg.get(row1)).getWrapped());
+                                    assertEquals(16L, ((LongWrapper) yPlusAgg.get(row2)).getWrapped());
+                                    assertEquals(13L, ((LongWrapper) yPlusAgg.get(row3)).getWrapped());
+
+                                    // Expand row 1, exposing constituent rows
+                                    rollupTable.expand(JsTreeTable.RowReferenceUnion.of(1), null);
+                                    return rollupTable.<TreeViewportData>nextEvent(
+                                            JsTreeTable.EVENT_UPDATED, 2001d);
+                                }).then(event -> {
+                                    TreeViewportData data = event.getDetail();
+
+                                    Column yCol = rollupTable.findColumn("Y");
+                                    Column yPlusConst = rollupTable.findColumn("YPlusConst");
+
+                                    // Row 2 is the first constituent of row 1, with 7 constituent rows
+                                    for (int i = 2; i < 2 + 7; i++) {
+                                        TreeViewportData.TreeRow row =
+                                                (TreeViewportData.TreeRow) data.getRows().getAt(i);
+                                        assertFalse(row.isExpanded());
+                                        assertFalse(row.hasChildren());
+                                        assertEquals(3, row.depth());
+
+                                        // get the raw Y value and test against the plus 1
+                                        int yVal = yCol.get(row).asInt();
+                                        assertEquals(yVal + 1, yPlusConst.get(row).asInt());
+                                    }
+
+                                    rollupTable.close();
+                                    assertTrue(rollupTable.isClosed());
+                                    return null;
+                                });
+                    });
+                })
+                .then(this::finish).catch_(this::report);
+    }
+
+    public void testCreateRollupUpdateViewUnsafe() {
+        connect(tables)
+                .then(table("static_table_to_rollup"))
+                .then(table -> {
+                    JsRollupConfig cfg = new JsRollupConfig();
+                    cfg.groupingColumns = Js.uncheckedCast(JsArray.of("X"));
+                    cfg.includeConstituents = true;
+                    cfg.aggregations = JsPropertyMap.of(JsAggregationOperation.SUM, JsArray.of("Y"));
+                    return table.rollup(cfg).then(rollupTable -> {
+
+                        JsPropertyMap<Object> col0 =
+                                JsPropertyMap.of("name", "UnsafeOp", "expression", "Runtime.exec(`ls`)", "type",
```

Comment:
Probably should instead be
```suggestion
                                JsPropertyMap.of("name", "UnsafeOp", "expression", "Runtime.getRuntime().exec(`ls`)", "type",
```

Can you check quick that this actually passed before your validation was added, and now correctly fails? As written, I think you should get an error that exec() is not static.

---

This isn't necessarily actionable for the current ticket, but just getting it out there. You could consider changing the test from being quite so overt of a call to something more benign but still forbidden.

io.deephaven.server.table.validation.ColumnExpressionValidator is the class that is populated with the currently allowed methods - this is checked through a hardcoded set of instance and static methods.

Compare with io.deephaven.engine.table.lang.impl.QueryLibraryImportsDefaults, which provides the defaults in the QueryLibrary. Most of those statically imported methods are present in the validator, but not all, such as Color's `colorRGB(int,int,int)` method. 

But what you have now should protect us adequately (with a better test).